app/javascript/controllers/nested_roles_permission_controller.js

```diff
@@ -0,0 +1,28 @@
+import { Controller } from 'stimulus'
+
+export default class extends Controller {
+  static targets = [ 'targetOne', 'templateOne' ]
+  static values = { wrapperSelector: String }
+ 
+  initialize () {
+    this.wrapperSelector = this.wrapperSelectorValue || '.nested-roles-permission-wrapper'
+  }
+
+  addPermissions (e) {
+    e.preventDefault()
+
+    let contentOne = this.templateOneTarget.innerHTML.replace(/NEW_RECORD/g, new Date().getTime().toString())
+    this.targetOneTarget.insertAdjacentHTML('beforebegin', contentOne)
```

Comment:
To make ESLint happy, I think you need:
```
 this.targetOneTarget.insertAdjacentHTML('beforebegin', escapeHTML`=${contentOne}`)
```

From [this page](https://www.npmtrends.com/eslint-plugin-no-unsafe-innerhtml).